### PR TITLE
Adding element type page in from wiki

### PR DESCRIPTION
**Pull request by adamdewey**
Re-ordered a few menu items too to try and align more with https://diataxis.fr/

### CODE REVIEW

**Review by CarlosNZ — CHANGES_REQUESTED** [review]
Cool, thanks Adam, will have a proper look soon. Would you also mind updating the page in the original page in the dev docs to just have a note linking to this page? (so we know to only maintain one page going forward)

Should be able to do that in this PR too.

**Comment by adamdewey** [comment]
> Cool, thanks Adam, will have a proper look soon. Would you also mind updating the page in the original page in the dev docs to just have a note linking to this page? (so we know to only maintain one page going forward)
> 
> Should be able to do that in this PR too.

Sure - not quite sure how I update a PR though?

**Comment by CarlosNZ** [comment]
> Sure - not quite sure how I update a PR though?

Just make another commit on this branch and push it, that's all :)

**Comment by adamdewey** [comment]
> > Sure - not quite sure how I update a PR though?
> 
> Just make another commit on this branch and push it, that's all :)

Thanks Carl - for some reason GitHub Desktop kept telling me I was making dirty subproject changes, so I just updated the wiki using the UI!

**Comment by CarlosNZ** [comment]
>  so I just updated the wiki using the UI!

That won't work -- as soon as someone makes a change to the internal docs, the "wiki" page will get over-written next update. (Docs -> wiki changes should only go in one direction)

You need to update `Element-Type-Specs.md` in /docs/internal. Don't worry about making it update the wiki just yet. Our normal workflow is to update the docs, then when it's approved/merged update the wiki (which we do using the `yarn push_docs` command).

If you're getting dirty subproject errors, that just means your submodule (the _wiki folder) doesn't match the commit that that the "outer" repo has on record for it. You can ignore it, but it you want it to be correct, just go into the "_wiki" folder at the command line and use `git pull`. (If you want further clarification around this, just give me a call -- it took me ages to understand what's going on with the submodule commits).

**Comment by adamdewey** [comment]
> > so I just updated the wiki using the UI!
> 
> That won't work -- as soon as someone makes a change to the internal docs, the "wiki" page will get over-written next update. (Docs -> wiki changes should only go in one direction)
> 
> You need to update `Element-Type-Specs.md` in /docs/internal. Don't worry about making it update the wiki just yet. Our normal workflow is to update the docs, then when it's approved/merged update the wiki (which we do using the `yarn push_docs` command).
> 
> If you're getting dirty subproject errors, that just means your submodule (the _wiki folder) doesn't match the commit that that the "outer" repo has on record for it. You can ignore it, but it you want it to be correct, just go into the "_wiki" folder at the command line and use `git pull`. (If you want further clarification around this, just give me a call -- it took me ages to understand what's going on with the submodule commits).

Ah ok, gotcha - thanks for that, I was updating in /docs/_wiki which was why it was giving me the dirty. 

Have now updated /doc/internal 

Cheers

File: docs/internal/Element-Type-Specs.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+_**Note: this document has been superseded by the zola site https://docs.mflow.nz/**_
```
Comment (CarlosNZ) [inline docs/internal/Element-Type-Specs.md:1]:
```suggestion
_**Note: Element type reference is available in the [public docs](https://docs.mflow.nz/<put-the-exact-url-here-if-you-know-it>) **_
```

File: docs/internal/Element-Type-Specs.md [hunk 2]
```diff
@@ -1,3 +1,5 @@
...
+
 _Ongoing authoritative reference of Template Question/Element types, including input parameters and response type (shape). Please ensure this document matches the current implementation at all times._
```
Comment (CarlosNZ) [inline docs/internal/Element-Type-Specs.md:3]: Delete everything else -- no point have two slightly different versions hanging around.

File: docs/public/content/reference/forms/index.md [hunk 3]
```diff
@@ -0,0 +1,661 @@
...
+#### Input parameters
+
+- ~~**pageBreakValidityCheck**: `boolean` -- If `true`, the user cannot proceed to the next page unless _all_ questions on the current page have passed validation~~
+  - ~~default: `false`~~
+
+---
+
+```
+
+```
```
Comment (CarlosNZ) [inline docs/public/content/reference/forms/index.md:656]: This can go.

**Review by CarlosNZ — CHANGES_REQUESTED** [review]
Great, just a couple of tiny things.

File: docs/internal/Element-Type-Specs.md (hunk 2 above)
Comment (adamdewey) [reply docs/internal/Element-Type-Specs.md]: Cool, just deleted and committed

File: docs/public/content/reference/forms/index.md (hunk 3 above)
Comment (adamdewey) [reply docs/public/content/reference/forms/index.md]: This has gone also now

File: docs/internal/Element-Type-Specs.md [hunk 4]
```diff
@@ -1,626 +1,3 @@
-_Ongoing authoritative reference of Template Question/Element types, including input parameters and response type (shape). Please ensure this document matches the current implementation at all times._
+_**Note: Element type reference is available in the [public docs](https://docs.mflow.nz/<put-the-exact-url-here-if-you-know-it>) **_
```
Comment (CarlosNZ) [inline docs/internal/Element-Type-Specs.md:1]: `https://docs.mflow.nz/<put-the-exact-url-here-if-you-know-it>`

Do you know it? Well -- do ya?

Comment (adamdewey) [reply docs/internal/Element-Type-Specs.md]: Oh! Haha, should be there now

**Review by CarlosNZ — APPROVED** [review]
Nice :)

**Pull request merged**